gateway/ln-gateway/src/rpc/lnrpc_client.rs

```diff
@@ -142,20 +144,81 @@ impl ILnRpcClient for NetworkLnRpcClient {
     async fn subscribe_intercept_htlcs(
         &self,
         short_channel_id: u64,
-    ) -> Result<mpsc::Receiver<HtlcInterceptPayload>> {
+    ) -> Result<mpsc::Receiver<GatewayChannelMessage>> {
         let mut client = self.client.lock().await.clone();
-
-        let mut response_stream = client
+        let mut stream = client
             .subscribe_intercept_htlcs(Request::new(SubscribeInterceptHtlcsRequest {
                 short_channel_id,
             }))
             .await
             .expect("Failed to subscribe intercept htlcs")
             .into_inner();
 
-        let (_, rx) = mpsc::channel::<HtlcInterceptPayload>(100);
-
-        // TODO: Send intercepted HTLCs from stream to gateway actors for processing
+        // Spawn a task to listen for messages from the htlc stream.
+        // Adapt HTLC data from protobuf types to gateway types and send for processing.
+        let mut tg = self.task_group.clone();
+        let (tx, rx) = mpsc::channel::<GatewayChannelMessage>(100);
+
+        tg.spawn(
```

Comment:
I assume `subscribe_intercept_htlcs` is called a finite, config-defined, small amount of times? Iirc our task group implementation does not handle the spawning of ephemeral tasks, so its data structures would fill up over time and overflow. @dpc is this still correct?

Reply:
`ts.make_subgroup` should handle it.